private/bufpkg/bufcheck/bufcheck.go

```diff
@@ -162,27 +164,44 @@ func WithPluginsEnabled() ClientFunctionOption {
 	return pluginsEnabledOption{}
 }
 
-// RunnerProvider provides pluginrpc.Runners for program names and args.
-type RunnerProvider interface {
-	NewRunner(programName string, programArgs ...string) pluginrpc.Runner
+// Plugin is a configured lint or breaking plugin that wraps a pluginrpc.Runner.
+type Plugin interface {
```

Comment:
I don't fully understand this interface - it doesn't make sense to me why a Plugin is a Runner. Should I be able to execute arbitrary arguments on a Plugin? Almost for sure no.

Reply:
A plugin is run/invoked so I think having a method Run makes sense. Although maybe the interface name shouldn't be `Runner` as that does imply handling multiple.

Reply:
I probably should have been more direct, apologies - I don't think the interface signature makes sense here. I'm not super convinced by this change as a whole, wondering if your RunnerProvider idea might be less invasive.